packages/shared/src/components/plus/GiftPlusModal.tsx

```diff
@@ -0,0 +1,274 @@
+import classed from '../../lib/classed';
+import JoinedDate from '../profile/JoinedDate';
+import { webappUrl } from '../../lib/constants';
+
+interface GiftPlusModalProps extends ModalProps {
+  preselected?: UserShortProfile;
+  onSelected?: (user: UserShortProfile) => void;
+}
+
+const UserText = classed('span', 'flex flex-row items-center justify-center');
+
+export function GiftPlusModalComponent({
+  preselected,
+  onSelected,
+  ...props
+}: GiftPlusModalProps): ReactElement {
+  const [overlay, setOverlay] = useState<HTMLElement>();
+  const { onRequestClose } = props;
+  const { giftOneYear } = usePaymentContext();
+  const [selected, setSelected] = useState(preselected);
+  const [index, setIndex] = useState(0);
+  const [query, setQuery] = useState('');
+  const [onSearch] = useDebounceFn(setQuery, 500);
+  const { data: users } = useQuery<UserShortProfile[]>({
+    queryKey: ['search', 'users', query],
+    queryFn: async () => {
+      const result = await gqlClient.request(RECOMMEND_MENTIONS_QUERY, {
+        query,
+      });
+
+      setIndex(0);
+
+      return result.recommendedMentions;
+    },
+    enabled: !!query?.length,
+  });
+  const isVisible = !!users?.length && !!query?.length;
+  const onKeyDown = (e: React.KeyboardEvent) => {
+    const movement = [ArrowKey.Up, ArrowKey.Down, KeyboardCommand.Enter];
+    if (!movement.includes(e.key as (typeof movement)[number])) {
+      return;
+    }
+
+    if (!users?.length) {
+      return;
+    }
+
+    e.preventDefault();
+
+    if (e.key === ArrowKey.Down) {
+      setIndex((prev) => {
+        let next = prev + 1;
+        let counter = 0;
+        while (users[next % users.length]?.isPlus) {
+          next += 1;
+          counter += 1;
+
+          if (counter > users.length) {
+            return -1;
+          }
+        }
+        return next % users.length;
+      });
+    } else if (e.key === ArrowKey.Up) {
+      setIndex((prev) => {
+        let next = prev - 1;
+        let counter = 0;
+        while (users[(next + users.length) % users.length]?.isPlus) {
+          next -= 1;
+          counter += 1;
+
+          if (counter > users.length) {
+            return -1;
+          }
+        }
+        return (next + users.length) % users.length;
+      });
+    } else {
+      setSelected(users[index]);
+    }
+  };
+
+  const onSelect = (user: UserShortProfile) => {
+    setSelected(user);
+    setIndex(0);
+    setQuery('');
+  };
+
+  const isTablet = useViewSize(ViewSize.Tablet);
+
+  return (
+    <Modal
+      {...props}
+      isOpen
+      kind={Modal.Kind.FixedCenter}
+      size={Modal.Size.Small}
+      overlayRef={setOverlay}
+      isDrawerOnMobile
+    >
+      <Modal.Body className="gap-4 tablet:!p-4">
+        <div className="flex flex-row justify-between">
+          <PlusTitle type={TypographyType.Callout} bold />
+          <CloseButton
+            type="button"
+            size={ButtonSize.Small}
```

Comment:
Isn't closeButton always the same everywhere?

Reply:
We have varying sizes depending on where it was used.